PWGHF/TableProducer/candidateSelectorD0.cxx

```diff
@@ -357,15 +360,7 @@ struct HfCandidateSelectorD0 {
 
       if (applyMl) {
         // ML selections
-
-        std::vector<float> inputFeatures{candidate.cpa(),
-                                         candidate.cpaXY(),
-                                         candidate.decayLength(),
-                                         candidate.decayLengthXY(),
-                                         candidate.impactParameter0(),
-                                         candidate.impactParameter1(),
-                                         candidate.impactParameterProduct()};
-
+        std::vector<float> inputFeatures = hfMlResponse.getInputFeatures(candidate, trackPos, trackNeg);
```

Comment:
In the hard-coded version we did not have variables that depend on the mass hypothesis. However, now that we have the flexibility to modify the variables via the configurable, we should differentiate the two. In particular, if the candidate is preselected as a `D0toPiK` (`statusD0`) we should get the BDT output scores corresponding to that mass hypothesis, while if selected as `D0bartoKPi` (`statusD0bar`) the other ones. This means that sometimes the `isSelectedMl` method will be called for one of the two, sometimes for both. We should then decide a convention how to store them in the `hfMlD0Candidate` table (@fcatalan92 any idea?)

Reply:
Yes this might indeed be a possibility. In this case we should decide whether to fill the columns with empty vectors in case there are no BDT output scores for one of the hypothesis (based on the `statusD0` or `statusD0bar` before the BDT application) or to always test the BDT for both hypotheses and fill it.